deployment/administration/SRE_Delete_Unassigned_Users.ps1

```diff
@@ -0,0 +1,46 @@
+param(
+    [Parameter(Mandatory = $true, HelpMessage = "Enter SHM ID (e.g. use 'testa' for Turing Development Safe Haven A)")]
+    [string]$shmId
+)
+
+Import-Module Az.Accounts -ErrorAction Stop
+Import-Module $PSScriptRoot/../common/AzureCompute -Force -ErrorAction Stop
+Import-Module $PSScriptRoot/../common/Configuration -Force -ErrorAction Stop
+Import-Module $PSScriptRoot/../common/Logging -Force -ErrorAction Stop
+
+# Get config and original context 
+# -------------------------------
+$config = Get-ShmConfig -shmId $shmId
+$originalContext = Get-AzContext
+
+# Extract list of users
+# ---------------------
+$null = Set-AzContext -SubscriptionId $config.subscriptionName -ErrorAction Stop
+Add-LogMessage -Level Info "Exporting user list for $($config.shm.id) from $($config.dc.vmName)..."
+# Run remote script
+$script = @"
+`$userOuPath = (Get-ADObject -Filter * | Where-Object { `$_.Name -eq "Safe Haven Research Users" }).DistinguishedName
+`$users = Get-ADUser -Filter * -SearchBase "`$userOuPath" -Properties *
+foreach (`$user in `$users) {
+    `$groupName = (`$user | Select-Object -ExpandProperty MemberOf | ForEach-Object { ((`$_ -Split ",")[0] -Split "=")[1] }) -join "|"
+    `$user | Add-Member -NotePropertyName GroupName -NotePropertyValue `$groupName -Force
+}
```

Comment:
Not quite sure of the logic here. You combine all groups that the user is a member of with `|` (eg. `group1|group2`) then add this as a `GroupName` property on each user?

I don't think you need to be writing properties to users for this to work.